src/types.dom.ts

```diff
@@ -293,7 +293,8 @@ export interface RequestInit {
   referrerPolicy?: ReferrerPolicy
   signal?: AbortSignal | null
   timeout?: number
-  window?: any
+  window?: any,
+  requestFormatter?: (payload: RequestInit) => Promise<RequestInit> | RequestInit;
```

Comment:
I would suggest here making the accepted type be `RequestFormatter | RequestFormatter[]` and make the field name `requestFormatters` (plural). I would treat the singular form as sugar for passing an array of one.

Reply:
I think we should break the indirect recursion. Instead of `RequestInit` being the type returned by a `RequestFormatter` I think it should be a subset, data only. We could extract a new type called `RequestInitData`. I'm not sure but maybe it can have everything except `window` and `requestFormatter` which are clearly not data.